### PR TITLE
feat/additional-qol

### PR DESCRIPTION
**Pull request by Gidsss**
closes issue #314, #313, #317 

Keyboard Shortcuts:
F3 - Auto Format
F4 - Quick Run
F5 - Compile and Run
F11 - Exit Fullscreen
Ctrl + S - Save File
Ctrl + O - Open File

### CODE REVIEW

File: src/uwu.py [hunk 1]
```diff
@@ -158,8 +168,27 @@ def on_compile_and_run(self, code_editor: CodeEditor):
...
+        if len(code_editor.lx_errors) > 0:
+            self.analyzer_panel.clear_parser_tree()
+            self.code_panel.update_error_logs(errors=code_editor.lx_errors)
+        elif len(code_editor.p_errors) > 0:
+            self.code_panel.update_error_logs(errors=code_editor.p_errors)
+        elif len(code_editor.a_errors) > 0:
+            self.code_panel.update_error_logs(errors=code_editor.a_errors)
+        else:
+            self.code_panel.update_error_logs(errors=[])
+
```
Comment (chaaals) [inline src/uwu.py:188]: may not be necessary, `on_compiler_run` can have an extra parameter to use quick mode para hindi `code_editor.start()` ang gamitin niya

File: src/components/code_view.py [hunk 2]
```diff
@@ -358,7 +361,33 @@ def __init__(self, master, parent, **kwargs):
...
+            source_code = active_editor.text.get('1.0', 'end-1c')  # Fetching the latest code from the editor
+            active_editor.update_source_code(source_code)  # Method to update the editor's source code attribute if needed
+
+            # Run lexer, parser, and analyzer every time to reflect latest changes
+            lx_res = active_editor.run_lexer()
+            p_res = active_editor.run_parser()
+            a_res = active_editor.run_analyzer()
+
+            # Update analyzer tabs and compiler logs
+            self.parent.master.update_analyzer_tabs_and_logs(active_editor, lx_res, p_res, a_res)
```
Comment (chaaals) [inline src/components/code_view.py:378]: if implemented correctly, `on_compiler_run` should update the tabs and logs na, this won't be necessary

File: src/components/command_menu.py [hunk 3]
```diff
@@ -31,7 +31,20 @@ def __init__(self, master, parent, code_view: CodeView, **kwargs):
...
+        self.quickRunButtonBgImage = CTkImage(light_image=Image.open(f"{RUN_ASSET}")) # Change the image asset tomorrow
+        self.quickRunButton = CTkButton(
+            master=self,
+            image=self.quickRunButtonBgImage,
+            fg_color='#1A1B26',
+            text='Quick Run',
+            font=('JetBrains Mono', 12),
+            width=99,
+            height=30,
+            command=self.code_view.quick_run
```
Comment (chaaals) [inline src/components/command_menu.py:45]: e.g. `self.code_view.on_compile_and_run(compile=false)`

File: src/components/code_view.py [hunk 4]
```diff
@@ -358,7 +361,33 @@ def __init__(self, master, parent, **kwargs):
...
+
+            if lx_res and p_res and a_res:
+                if active_editor.transpiled_program:
+                    compiler = Compiler(py_source=active_editor.transpiled_program, filename=active_editor.filename)
+                    compiler.run_python()  # Run the Python transpiled code
```
Comment (chaaals) [inline src/components/code_view.py:383]: this is correct, `quick_run` should be getting the transpiled program, verifying, and running it sa cmd.

evaluation should be done before this is called. try to limit na ang nangyayari lang sa quick run is this

File: src/components/code_view.py [hunk 5]
```diff
@@ -358,7 +361,33 @@ def __init__(self, master, parent, **kwargs):
         for file in self.file_names:
             self.create_new_tab(file)
             self.bind_esc(editor=self.code_editors[file], file_name=file)
-    
+
+    def quick_run(self):
```
Comment (chaaals) [inline src/components/code_view.py:365]: consider moving this to `CodeEditor`, andon kasi lahat ng pag run ng program, at that level no need i-check updated source code

**Review by chaaals — CHANGES_REQUESTED** [review]
Sorry if madami, ideally with these changes, it will follow the same implementation we have on `Compile & Run`. Dapat nag i-iswitch lang yung mode ng `on_compile_and_run` to quick or compile modes

**Comment by Gidsss** [comment]
image
Demo:
https://github.com/Gidsss/UwUIDE/assets/88811389/091cce8b-4ae6-4f76-9f06-0b033d39bc94

**Review by am-cid — APPROVED** [review]

File: src/components/code_view.py [hunk 6]
```diff
@@ -410,12 +414,11 @@ def load_file(self):
             self.editor.init_linenums()
 
     def auto_format_code(self):
-
         # Validate source code
         lx_res = self.editor.run_lexer()
         p_res = self.editor.run_parser()
 
-        self.parent.code_panel.update_compiler_logs(editor=self.editor, is_compiling=False, is_formatting=True)
+        self.parent.master.code_panel.update_compiler_logs(editor=self.editor, is_compiling=False, is_formatting=True)
```
Comment (chaaals) [inline src/components/code_view.py:421]: why this change?

**Review by chaaals — CHANGES_REQUESTED** [review]
Noticed that lines 421, 433, 438, and 440 in `src/components/code_view.py` had these changes, why?

File: src/components/code_view.py (hunk 6 above)
Comment (Gidsss) [reply src/components/code_view.py:421]: my bad g hahahaha

**Review by HolyShaq — APPROVED** [review]

**Review by chaaals — APPROVED** [review]

**Pull request merged**